src/lepy/lego.go

```diff
@@ -0,0 +1,178 @@
+package main
+
+import "C"
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"encoding/json"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/go-acme/lego/v4/certcrypto"
+	"github.com/go-acme/lego/v4/certificate"
+	"github.com/go-acme/lego/v4/challenge/http01"
+	"github.com/go-acme/lego/v4/challenge/tlsalpn01"
+	"github.com/go-acme/lego/v4/lego"
+	"github.com/go-acme/lego/v4/providers/dns"
+	"github.com/go-acme/lego/v4/registration"
+)
+
+type LegoInputArgs struct {
+	Email  string `json:"email"`
+	Server string `json:"server"`
+	CSR    string `json:"csr"`
+	Plugin string `json:"plugin"`
+	Env    map[string]string
+}
+
+type LegoOutputResponse struct {
+	CSR               string `json:"csr"`
+	PrivateKey        string `json:"private_key"`
+	Certificate       string `json:"certificate"`
+	IssuerCertificate string `json:"issuer_certificate"`
+	Metadata          `json:"metadata"`
+}
+
+type Metadata struct {
+	StableURL string `json:"stable_url"`
+	URL       string `json:"url"`
+	Domain    string `json:"domain"`
+}
+
+//export RunLegoCommand
+func RunLegoCommand(message *C.char) *C.char {
+	CLIArgs, err := extractArguments(C.GoString(message))
+	if err != nil {
+		return C.CString(fmt.Sprint("error: couldn't extract arguments: ", err))
+	}
+	for k, v := range CLIArgs.Env {
+		os.Setenv(k, v)
```

Comment:
`Setenv` can return an error and it is unchecked.

Maybe it would be a good idea to also add `golangci-lint` in this project to catch things like this.

Reply:
added an errcheck.